staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
it's more that I want users to take care announcing/registering api groups and not randomly call them in a bunch of places "to make sure". I'd rather have a way to import the client that allows you to manually register. I know this is an intermediate state and once we give the client its own copy of the groups, we won't have this problem any more, but I worry that if we stay in the intermediate state for more than a few hours people will add double registrations all over and it will be hard to clean up the mess.